### PR TITLE
ORCA: Explicitly fallback to planner if target list is empty

### PR DESCRIPTION
**Pull request by l-wang**
Co-authored-by: Jesse Zhang <sbjesse@gmail.com>

cherry-picked
https://github.com/greenplum-db/gpdb-postgres-merge/commit/98154a8f6c47fc6007638ffe84e7ae8999b19f2a

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by gpopt — APPROVED** [review]
+1

**Review by chrishajas — APPROVED** [review]
Is there a test case we can add?

File: src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp [hunk 1]
```diff
@@ -2874,6 +2874,39 @@ CTranslatorQueryToDXL::TranslateSetOpChild(Node *child_node,
 	GPOS_ASSERT(NULL != colids);
 	GPOS_ASSERT(NULL != input_col_mdids);
 
+	// GPDB_12_MERGE_FIXME: We have to fallback here because otherwise we trip
```
Comment (gpopt) [inline src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp:2877]: Maybe we can remove `GPDB_12_MERGE_FIXME: `.

Comment (l-wang) [reply src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp:2877]: Yes will do.

**Comment by l-wang** [comment]
> Is there a test case we can add?

We have tests on 7X that came with Postgres 12, but no tests on 6X. Thank you for asking, I will back port the tests as well.

**Review by bhuvnesh2703 — APPROVED** [review]

**Pull request merged**